hack/tools/release/docker-image-verify.sh

```diff
@@ -0,0 +1,58 @@
+#!/usr/bin/env bash
+# Copyright 2023 The Kubernetes Authors.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+set -o errexit
+set -o nounset
+set -o pipefail
+
+_tmp="$(pwd)/_tmp"
+cleanup() {
+  rm -rf "${_tmp}"
+}
+trap "cleanup" EXIT SIGINT
+cleanup
+mkdir -p "${_tmp}"
+
+function TESTIMAGE() {
+  IMAGE="${1}"
+  ARCH="${2}"
+  GREPGOPATH="${3}"
+  BINARYPATH="${4:-manager}"
+
+  echo "> Testing $IMAGE"
+
+  docker save "${IMAGE}" -o "${_tmp}"/img.tar
+  mkdir -p "${_tmp}"/extracted-img "${_tmp}"/extracted
+  tar -xf "${_tmp}"/img.tar -C "${_tmp}"/extracted-img/
+  while IFS= read -r -d '' layer
+  do
+    tar -xf "${layer}" -C "${_tmp}"/extracted
+  done <   <(find "${_tmp}"/extracted-img/ -name "*.tar" -print0)
+
+  go version -m "${_tmp}"/extracted/"${BINARYPATH}" | grep -E $'\tpath' | grep -E -q -e "${GREPGOPATH}" || (echo "FAILED ${IMAGE} expected value for path: \"${GREPGOPATH}\""; go version -m "${_tmp}"/extracted/"${BINARYPATH}" | grep -E $'\tpath'; exit 1)
+  go version -m "${_tmp}"/extracted/"${BINARYPATH}" | grep -q -E "GOARCH=${ARCH}$" || (echo "Failed ${IMAGE} expected GOARCH to be \"$ARCH\""; go version -m "${_tmp}"/extracted/"${BINARYPATH}" | grep "GOARCH="; exit 1)
+
+  rm -rf "${_tmp}"/img.tar "${_tmp}"/extracted-img "${_tmp}"/extracted
+}
+
+for arch in ${ALL_ARCH}; do
+  TESTIMAGE "${REGISTRY}/cluster-api-controller-${arch}:${TAG}" "${arch}" "sigs.k8s.io/cluster-api$"
+  TESTIMAGE "${REGISTRY}/kubeadm-bootstrap-controller-${arch}:${TAG}" "${arch}" "sigs.k8s.io/cluster-api/bootstrap/kubeadm$"
+  TESTIMAGE "${REGISTRY}/kubeadm-control-plane-controller-${arch}:${TAG}" "${arch}" "sigs.k8s.io/cluster-api/controlplane/kubeadm$"
+  TESTIMAGE "${REGISTRY}/capd-manager-${arch}:${TAG}" "${arch}" "command-line-arguments$"
```

Comment:
@chrischdi Not super important. Do you have any idea why in CAPD we have command-line-arguments?

Reply:
Maybe due to difference between building both binaries.

CAPD build:
```
# Build the CAPD manager using the compiler cache folder
RUN --mount=type=cache,target=/root/.cache/go-build \
    --mount=type=cache,target=/go/pkg/mod \
    CGO_ENABLED=0 GOOS=linux GOARCH=${ARCH} go build -trimpath -a -o /workspace/manager main.go
```

and

Core controller:
```
# Do not force rebuild of up-to-date packages (do not use -a) and use the compiler cache folder
RUN --mount=type=cache,target=/root/.cache/go-build \
    --mount=type=cache,target=/go/pkg/mod \
    CGO_ENABLED=0 GOOS=linux GOARCH=${ARCH} \
    go build -trimpath -ldflags "${ldflags} -extldflags '-static'" \
    -o manager ${package}
```

Not sure if this is the actual reason.

Reply:
The reason is in CAPD we refer a file `main.go` and for the other controllers we pass the package.

Changing `main.go` to `.` (to refer the relative package inside the workdir) would fix the path also for capd: `        path    sigs.k8s.io/cluster-api/test/infrastructure/docker`

Reply:
Maybe worth doing that change and change the test to `sigs.k8s.io/cluster-api/test/infrastructure/docker` ?

Reply:
Yeah. Let's do it in a separate PR though. Can one of you follow-up please?

Reply:
sure, I will raise a separate PR directly as it is small change.

Reply:
@akshay196 Just fyi @chrischdi already opened the follow-up PR (#10030). So we're good :)